core/dbt/graph/selector_methods.py

```diff
@@ -567,17 +591,24 @@ def search(self, included_nodes: Set[UniqueId], selector: str) -> Iterator[Uniqu
         manifest: WritableManifest = self.previous_state.manifest
 
         for node, real_node in self.all_nodes(included_nodes):
```

Comment:
what do you think about refactoring this bit to look more like: 
```python
        for node, real_node in self.all_nodes(included_nodes):
            previous_node: Optional[SelectorTarget] = None

            if node in manifest.nodes:
                previous_node = manifest.nodes[node]
            elif node in manifest.sources:
                previous_node = manifest.sources[node]
            elif node in manifest.exposures:
                previous_node = manifest.exposures[node]
            elif node in manifest.metrics:
                previous_node = manifest.metrics[node]

            keyword_args = {}
            if checker.__name__ in ["same_contract", "check_modified_content"]:
                keyword_args["adapter_type"] = adapter_type  # type: ignore

            if checker(previous_node, real_node, **keyword_args):  # type: ignore
                yield node
```

My thinking is that it's more clear / readable to isolate the `adapter_type` optionality in keyword_args from setting `previous_node` and `real_node`. I tested it out locally but could be missing something on why it wouldn't work!

Reply:
This is better!